practice/completer/site.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+/**
+ *  The Completer Constructor holds valid words that can be called through
+ * the prototype method complete, to find valid words that start with given
+ * prefix.
+ * There is also a weighting system, where everytime selectCompletion is called
+ * the selected string has weight added to it. The complete function returns
+ * an array ordered in more weighted words first.
+ */
+function Completer() {
+  this.validCompletionToWeight = {};
+}
+
+var completerProto = {
+  addCompletion: function(str) {
+    this.validCompletionToWeight[str] = 0;
+  },
+  removeCompletion: function(str) {
+    delete this.validCompletionToWeight[str];
+  },
+  complete: function(prefix) {
+    var potentialWords = _.filter(
+      _.keys(this.validCompletionToWeight), function(o) {
```

Comment:
I guess I lost my way here.  I thought the \_.keys() function strips off the keys from the object and returns an array of just the keys.  But that would eliminate the weights.  I guess my lodash ignorance is showing through...